core/vm/contract.go

```diff
@@ -49,24 +49,21 @@ type Contract struct {
 	caller        ContractRef
 	self          ContractRef
 
-	jumpdests destinations // result of JUMPDEST analysis.
+	jumpdests destinations // Aggregated result of JUMPDEST analysis.
+	analysis  bitvec       // Locally cached result of JUMPDEST analysis
 
 	Code     []byte
-	CodeHash common.Hash
+	CodeHash *common.Hash
```

Comment:
Wouldn't it be the same to keep this as it was and instead of checking for `nil`, we check for `== (common.Hash{})`?